google/cloud/storage/examples/storage_notification_samples.cc

```diff
@@ -54,9 +54,14 @@ void ListNotifications(google::cloud::storage::Client client, int& argc,
   namespace gcs = google::cloud::storage;
   [](gcs::Client client, std::string bucket_name) {
     std::cout << "Notifications for bucket=" << bucket_name << std::endl;
-    std::vector<gcs::NotificationMetadata> items =
-        client.ListNotifications(bucket_name).value();
-    for (gcs::NotificationMetadata const& notification : items) {
+    gcs::StatusOr<std::vector<gcs::NotificationMetadata>> status =
```

Comment:
I'm never sure how to name these StatusOr<T> thingies, but it might be better as `items` rather than `status`.  Then you have, ...
```
if (not items.ok()) ...  // reasonable
for (gcs::NotificationMetadata const& notification : *items) ...  // much nicer
```

Reply:
`*status` would be more readable with StatusOr if it used `status.value()` because it looks like magic otherwise. Not sure if this is common convention `*status` though.

Reply:
I went with `items`. @frankyn the `*items` convention is based on other standard C++ types, `optional<T>` has `operator*` for example, and most smart pointers do. After checking that `items.ok() == true` as in this example, the `operator*` and `operator->` are more efficient too.

Magic is what C++ is all about :grin: